app/controllers/schools/meters_controller.rb

```diff
@@ -77,17 +82,12 @@ def destroy
       redirect_to school_meters_path(@school)
     end
 
-  private
+    private
 
     def set_n3rgy_status
       return unless can?(:view_dcc_data, @school)
-      service = Meters::N3rgyMeteringService.new(@meter)
-      @known_to_n3rgy = service.available?
-      if @known_to_n3rgy && @meter.dcc_meter
-        @n3rgy_status = service.status
-        @n3rgy_consent_confirmed = service.consented?
-        @available_cache_range = service.available_data if @n3rgy_consent_confirmed
-      end
+
+      @n3rgy = Meters::N3rgyMeteringService.new(@meter)
```

Comment:
The original code was trying to reduce number of API calls. I think moving this into the view means we're doing more calls again?. Original: one for available, then consented? if known to n3rgy, and then we only get cache if consented. New view code calls available, consented? (which also calls available again), then consented? again (which calls available again) before getting the cache.

Suggest either adding the vars back here or some caching in the service.

Reply:
caching added